Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/tasks/data/source/InMemoryLocalTaskDataSource.java

```diff
@@ -82,24 +82,16 @@ public void call(Subscriber<? super SyncedData<Task>> subscriber) {
         }).compose(this.<SyncedData<Task>>delay());
     }
 
-    @Override
-    public Observable<Void> deleteAllTasks() {
-        return Observable.create(new Observable.OnSubscribe<Void>() {
-            @Override
-            public void call(Subscriber<? super Void> subscriber) {
-                localTasks = Tasks.empty();
-                subscriber.onCompleted();
-            }
-        }).compose(this.<Void>delay());
-    }
-
     private <T> Observable.Transformer<T, T> delay() {
         return new Observable.Transformer<T, T>() {
             @Override
             public Observable<T> call(Observable<T> observable) {
                 return observable.doOnEach(new Action1<Notification<? super T>>() {
                     @Override
                     public void call(Notification<? super T> t) {
+                        if (t.getKind() == Notification.Kind.OnCompleted) {
```

Comment:
Sorry @Dorvaryn I don't understand what you mean. Are you saying this is close to the actual API call behaviour  in that it will return `onNext` and `onComplete` at almost the same time rather than having an `onNext` and then delaying the emission of the `onComplete`?